src/rekapi.actor.js

```diff
@@ -90,70 +55,32 @@ rekapiModules.push(function (context) {
-      var i = 0, len = propertyTrack.length, keyframeProperty;
-
-      for (i; i < len; i++) {
-        keyframeProperty = propertyTrack[i];
-
-        if (keyframeProperty.millisecond > forMillisecond) {
-          // We went to far, use the previous keyframeProperty
-          latestProperties[propertyName] = previousKeyframeProperty;
-        } else if (keyframeProperty.millisecond === forMillisecond) {
-          // Found it!
-          latestProperties[propertyName] = keyframeProperty;
-        }
-
-        previousKeyframeProperty = keyframeProperty;
-
-        // Quit the loop if something was found.  We can't early-return above,
-        // because latestProperties[propertyName] might be null, which is not
-        // what we want.
-        if (latestProperties[propertyName]) {
-          break;
-        }
-      }
-
-      // If nothing was found, attempt to use the last keyframeProperty in the
-      // track.
-      if (!latestProperties[propertyName]) {
-        var lastProp = _.last(propertyTrack);
-
-        if (lastProp && lastProp.millisecond <= forMillisecond) {
-          latestProperties[propertyName] = lastProp;
-        }
+      var index = insertionPointInTrack(propertyTrack, forMillisecond);
+      if (propertyTrack[index] && propertyTrack[index].millisecond == forMillisecond) {
+        // Found forMillisecond exactly.
+        latestProperties[propertyName] = propertyTrack[index];
+      } else if (index >= 1) {
+        // forMillisecond doesn't exist in the track and index is
+        // where we'd need to insert it, therefore the previous
+        // keyframe is the most recent one before forMillisecond.
+        latestProperties[propertyName] = propertyTrack[index - 1];
+      } else {
+        // Return first property.  This is after forMillisecond.
+        latestProperties[propertyName] = propertyTrack[0];
       }
     });
 
     return latestProperties;
   }
 
-  /*!
-   * Links each KeyframeProperty to the next one in its respective track.
-   *
-   * They're linked lists!
-   * @param {Rekapi.Actor} actor
-   */
-  function linkTrackedProperties (actor) {
-    _.each(actor._propertyTracks, function (propertyTrack) {
-      _.each(propertyTrack, function (keyframeProperty, i) {
-        keyframeProperty.linkToNext(propertyTrack[i + 1]);
-      });
-    });
+  function insertionPointInTrack (track, millisecond) {
+    return _.sortedIndex(track, { millisecond: millisecond }, 'millisecond');
   }
 
-  /*!
-   * Returns a requested KeyframeProperty at a millisecond on a specified
-   * track.
-   * @param {Rekapi.Actor} actor
-   * @param {string} trackName
-   * @param {number} millisecond
-   * @return {Rekapi.KeyframeProperty|undefined}
-   */
-  function findPropertyAtMillisecondInTrack (actor, trackName, millisecond) {
-    return _.findWhere(actor._propertyTracks[trackName], {
-      millisecond: millisecond
-    });
+  function propertyIndexInTrack (track, millisecond) {
```

Comment:
It might be clearer to return `-1` if the property is not found.  It's more conventional, anyways.  What do you think?  Or is returning `undefined` more performant?